### PR TITLE
Error cleanly on missing Nulecule or invalid formatted Nulecule

### PR DESCRIPTION
**Pull request by cdrage**
This commit checks to see if the actual Nulecule exists before loading. If
the Nulecule file is missing it will cleanly error out.

If a Nulecule is wrongly formatted, atomicapp will exit cleanly with a line
number, column and the output of the invalid line.

ex.
``` 
2016-03-04 13:03:59,617 - [ERROR] - main.py - Failure parsing Nulecule
file. Validation error on line 12, column 23:
            {
```

Fixes issues:
#581 
#553

### CODE REVIEW

**Comment by cdrage** [comment]
Tests pass :+1:

File: atomicapp/nulecule/base.py [hunk 1]
```diff
@@ -128,10 +131,27 @@ def load_from_path(cls, src, config=None, namespace=GLOBAL_CONF,
             an image).
         """
         nulecule_path = os.path.join(src, MAIN_FILE)
+
+        if os.path.exists(nulecule_path):
+            nulecule_data = open(nulecule_path, 'r').read()
```
Comment (dustymabe) [inline atomicapp/nulecule/base.py]: should we use the contextlib stuff here:

```
with open() as f:
  nulecule_data = f.read()
```

I think that would be good practice.

**Comment by dustymabe** [comment]
@cdrage. nice work. I have a few questions:

- Do we think this is something that would actually be better to be implemented in Anymarkup itself, rather than in Atomic App? If so we could merge this like is, but open a PR on Anymarkup and let the functionality live there once it is merged.
- Would it be useful to give the full path to the file in the output? 
- Would it be useful to show more than one line of context and the line numbers on the left hand side?

**Comment by cdrage** [comment]
@dustymabe 

Yes, this should be put upstream, I'll send in a patch.

Full path :+1: 

I'll try that and see how it looks like.

**Comment by cdrage** [comment]
@dustymabe 
this looks better! :) updated the PR for another review.
```sh
▶ atomicapp run .
2016-03-09 09:11:46,115 - [INFO] - main.py - Action/Mode Selected is: run
2016-03-09 09:11:46,118 - [ERROR] - main.py - Failure parsing ./Nulecule file. Validation error on line 10, column 5:
9     - name: helloapache-app
10     params:
11       - name: image
```

**Comment by dustymabe** [comment]
#dotests

**Comment by dustymabe** [comment]
Nice work Charlie. The only concern I have is: is it possible that an error on the first or last line of the yaml file would throw off the output somehow. If that works just fine (and the tests pass that are in progress now) then LGTM.. merge away.

**Comment by cdrage** [comment]
Awesome :D I'll merge this in.

**Pull request merged**